pkg/stub/handler.go

```diff
@@ -862,6 +960,73 @@ func (h *Handler) processError(opcfg *v1alpha1.SamplesResource, ctype v1alpha1.S
 	return err
 }
 
+func (h *Handler) imageStream(imagestream, is *imagev1.ImageStream, opcfg *v1alpha1.SamplesResource) error {
+	if _, isok := h.skippedImagestreams[imagestream.Name]; !isok {
+		h.updateDockerPullSpec([]string{"docker.io", "registry.redhat.io", "registry.access.redhat.com", "quay.io"}, imagestream, opcfg)
+
+		if imagestream.Labels == nil {
+			imagestream.Labels = make(map[string]string)
+		}
+		imagestream.Labels[v1alpha1.SamplesManagedLabel] = "true"
+		if imagestream.Annotations == nil {
+			imagestream.Annotations = make(map[string]string)
+		}
+		imagestream.Annotations[v1alpha1.SamplesVersionAnnotation] = v1alpha1.CodeLevel
+
+		if is == nil {
+			_, err := h.imageclientwrapper.Create("openshift", imagestream)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "imagestream create error: %v")
+			}
+			logrus.Printf("created imagestream %s", imagestream.Name)
+		} else {
+			imagestream.ResourceVersion = is.ResourceVersion
+			_, err := h.imageclientwrapper.Update("openshift", imagestream)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "imagestream update error: %v")
+			}
+			logrus.Printf("updated imagestream %s", is.Name)
+		}
+	} else {
+		if is.Labels == nil {
+			is.Labels = make(map[string]string)
+		}
+		is.Labels[v1alpha1.SamplesManagedLabel] = "false"
+		h.imageclientwrapper.Update("openshift", is)
+		// if we get an error, we'll just try to remove the label next
+		// time; and we'll examine the skipped lists on delete
+	}
+	return nil
+}
+
+func (h *Handler) template(template, t *templatev1.Template, opcfg *v1alpha1.SamplesResource) error {
+	if _, tok := h.skippedTemplates[template.Name]; !tok {
+		template.Labels[v1alpha1.SamplesManagedLabel] = "true"
+		template.Annotations[v1alpha1.SamplesVersionAnnotation] = v1alpha1.CodeLevel
+
+		if t == nil {
+			_, err := h.templateclientwrapper.Create("openshift", template)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "template create error: %v")
+			}
+			logrus.Printf("created template %s", template.Name)
+		} else {
+			template.ResourceVersion = t.ResourceVersion
+			_, err := h.templateclientwrapper.Update("openshift", template)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "template update error: %v")
+			}
+			logrus.Printf("updated template %s", t.Name)
+		}
+	} else {
+		t.Labels[v1alpha1.SamplesManagedLabel] = "false"
+		h.templateclientwrapper.Update("openshift", t)
+		// if we get an error, we'll just try to remove the label next
+		// time; and we'll examine the skipped lists on delete
+	}
+	return nil
```

Comment:
idiomatic go (yeah i hate myself) for this structure (and the other method) should be:

```
if !skippedItem {
  do stuff
  return nil
}
// handle skippedItem logic
return nil
```

Or even better since the skippedItem logic is simpler, put that in the if block (ie minimize indentation).

Reply:
part of next push